test/opt/pass_fixture.h

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2016 Google Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and/or associated documentation files (the
+// "Materials"), to deal in the Materials without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Materials, and to
+// permit persons to whom the Materials are furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Materials.
+//
+// MODIFICATIONS TO THIS FILE MAY MEAN IT NO LONGER ACCURATELY REFLECTS
+// KHRONOS STANDARDS. THE UNMODIFIED, NORMATIVE VERSIONS OF KHRONOS
+// SPECIFICATIONS AND HEADER INFORMATION ARE LOCATED AT
+//    https://www.khronos.org/registry/
+//
+// THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+
+#ifndef LIBSPIRV_TEST_OPT_PASS_FIXTURE_H_
+#define LIBSPIRV_TEST_OPT_PASS_FIXTURE_H_
+
+#include <string>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include "opt/libspirv.hpp"
+#include "opt/pass_manager.h"
+#include "opt/passes.h"
+
+namespace spvtools {
+
+// Template class for testing passes. It contains some handy utility methods for
+// running passes and checking results.
+//
+// To write value-Parameterized tests:
+//   using ValueParamTest = PassTest<::testing::TestWithParam<std::string>>;
+// To use as normal fixture:
+//   using FixtureTest = PassTest<::testing::Test>;
+template <typename TestT>
+class PassTest : public TestT {
+ public:
+  PassTest() : tools_(SPV_ENV_UNIVERSAL_1_1) {}
+
+  // Adds a pass to be run.
+  template <typename PassT>
+  void AddPass() {
+    manager_.AddPass<PassT>();
+  }
+
+  // Runs the pass on the binary assembled from the |original| assembly, and
+  // checks whether the optimized binary can be disassembled to the |expected|
+  // assembly. Callers are suggested to use SCOPED_TRACE() for better messages.
+  void RunAndCheck(const std::string& original, const std::string& expected) {
+    assert(manager_.NumPasses());
```

Comment:
The regular assert might not trigger on release builds.  But ASSERT_TRUE would.
I am guessing here that you want to guard against accidentally making a test without passes. That's ok.